modvendor/modvendor_test.go

```diff
@@ -164,23 +1004,22 @@ func TestModVendorDoesNothingIfRefExists(t *testing.T) {
 }
 
 func TestModVendorNoRefFails(t *testing.T) {
-	// TODO(katcipis): when we start parsing modules for sources
-	// we need to address default remote references. For now it is
-	// always explicit.
-	const (
-		path = "github.com/mineiros-io/example"
-	)
-
 	s := sandbox.New(t)
 	gitURL := "file://" + s.RootDir()
 	rootdir := t.TempDir()
 
-	_, err := modvendor.Vendor(rootdir, "/vendor", tf.Source{
-		URL:  gitURL,
-		Path: path,
-	})
+	source, err := tf.ParseSource(fmt.Sprintf("git::%s", gitURL))
+	assert.NoError(t, err)
```

Comment:
If we go in the directionof always parsing and never creating tf.Source directly we could have a helper function to do this. Something like:

```suggestion
	source := parseSource(gitURL)
```

And it automatically adds the git:: prefix + fails if it is invalid etc.

Reply:
you mean:

```
source := parseSource(t, "git::%s?ref=%s", gitURL, ref)
```
right?

Reply:
Actually no, I think all formats are the same appending git::, so I was thinking on making it simpler and not spread around the whole formatting around the tests, it could even be something like newSource, then the formatting + parsing is just a detail to create the source correctly. Or we can leave as is, it is not bad either, just giving stupid ideas.